api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -18,27 +18,37 @@ package v1beta1
 
 import (
 	"fmt"
-	"reflect"
 
 	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/storage"
 	baremetalv1 "github.com/openstack-k8s-operators/openstack-baremetal-operator/api/v1beta1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// OpenStackDataPlaneRoleSpec defines the desired state of OpenStackDataPlaneRole
-type OpenStackDataPlaneRoleSpec struct {
+// OpenStackDataPlaneNodeSetSpec defines the desired state of OpenStackDataPlaneNodeSet
+type OpenStackDataPlaneNodeSetSpec struct {
 	// +kubebuilder:validation:Optional
 	// DataPlane name of OpenStackDataPlane for this role
 	DataPlane string `json:"dataPlane,omitempty"`
 
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
+	// AnsibleSSHPrivateKeySecret Private SSH Key secret containing private SSH
+	// key for connecting to node. Must be of the form:
+	// Secret.data.ssh-privatekey: <base64 encoded private key contents>
+	// <https://kubernetes.io/docs/concepts/configuration/secret/#ssh-authentication-secrets>
+	AnsibleSSHPrivateKeySecret string `json:"ansibleSSHPrivateKeySecret,omitempty"`
```

Comment:
Now this is defined here and in NodeTemplate? Same for NetworkConfig, and possibly some others.